cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -156,6 +161,46 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
+	@GetMapping("/assignment/{id}")
+	@Transactional
+	public Assignment getAssignment(@PathVariable int assignment_id) {
+		Assignment assignment = assignmentRepository.findById(assignment_id);
+		
+		return assignment;
+	}
+	
+	//As instructor, add a new assignment for my course.  The assignment has a name and a due date.
+	@PostMapping("/assignment")
+	@Transactional
+
+	public void addAssignment (@RequestParam String name, @RequestParam Date dueDate) {
+		
+		Assignment assignment = new Assignment();
+		assignment.setName(name);
+		assignment.setDueDate(dueDate);
+		assignmentRepository.save(assignment);
+	}
+	
+	
+	//As an instructor, I can change the name of the assignment for my course.
+	@PutMapping("/assignment/{id}")
+	@Transactional
+	public void changeAssignment (@PathVariable("id") Integer assignmentId, @PathVariable("name") String name ) {
```

Comment:
I am not sure that the second argument should be `@PathVariable("name")` then consumer facing URL becomes  like `/assignment/3/some-name` 
The name should probably be passed in the request body.